channeldb/graph.go

```diff
@@ -657,40 +672,137 @@ func (c *ChannelGraph) PruneGraph(spentOutputs []*wire.OutPoint,
 	return chansClosed, nil
 }
 
+// DisconnectBlockAtHeight is used to indicate that the block specified
+// by the passed heigth has been disconnected from the main chain. This
+// will "rewind" the graph back to the height below, deleting channels
+// that are no longer confirmed from the graph. The prune log will be
+// set to the last prune height valid for the remaining chain.
+func (c *ChannelGraph) DisconnectBlockAtHeight(height uint32) error {
```

Comment:
It'd be nice if this function returned all the channels that were _pruned_ just as the old `PruneGraph` does. This way we can notify to any of the the notification clients in the `ChannelRouter` that select channels were re-org'd out of the chain.